packages/nextjs-plugin/src/get-hashicorp-packages.ts

```diff
@@ -1,6 +1,25 @@
 import path from 'path'
 import fs from 'fs'
 
+function findUp(filename: string, start = process.cwd()) {
+  let result
+  const root = path.parse(start).root
+  let currentDir = start
+
+  while (!result || currentDir !== root) {
```

Comment:
I think this condition results in an infinite loop if the `filename` doesn't get found

What about
```typescript
while (!result) { 
  // ... still want to check the iteration when `currentDir === root`

  // at the end of the loop
  if (currentDir === root) break
}